src/backend/mod.rs

```diff
@@ -41,26 +41,27 @@ pub mod serial;
 
 #[cfg(all(
     target_arch = "x86_64",
-    any(feature = "simd_avx2", all(feature = "simd_avx512", nightly)),
     curve25519_dalek_bits = "64",
-    not(curve25519_dalek_backend = "fiat")
+    not(curve25519_dalek_backend = "fiat"),
+    not(curve25519_dalek_backend = "serial"),
 ))]
 pub mod vector;
 
 #[derive(Copy, Clone)]
 enum BackendKind {
     #[cfg(all(
         target_arch = "x86_64",
-        feature = "simd_avx2",
         curve25519_dalek_bits = "64",
-        not(curve25519_dalek_backend = "fiat")
+        not(curve25519_dalek_backend = "fiat"),
+        not(curve25519_dalek_backend = "serial")
```

Comment:
I think this file would be a lot easier to read if we just made all variants of this enum available. Does it actually matter whether or not `BackendKind::Avx2` exists on non-AVX2 machines? It is properly selected in any case, no?

Reply:
We can't use enum's in configuration predicates ? these happen in compile time where enum's are not available

Reply:
It gives dead code warning - I think in the long run we just may want to have proc macro for capability based selection.
I misread this originally - have fixed now.

Reply:
Oh it will give warnings about not covered cases for match - so we definitely need to keep it out of enum until we get a procmacro for this.

```
103 |     match get_selected_backend() {
    |           ^^^^^^^^^^^^^^^^^^^^^^ pattern `BackendKind::Avx512` not covered

```